packages/hoppscotch-app/newstore/MQTTSession.ts

```diff
@@ -0,0 +1,182 @@
+import { pluck, distinctUntilChanged } from "rxjs/operators"
+import DispatchingStore, { defineDispatchers } from "./DispatchingStore"
+
+type MQTTRequest = {
+  endpoint: string
+}
+
+type MQTTSession = {
+  request: MQTTRequest
+  connectingState: boolean
+  connectionState: boolean
+  subscriptionState: boolean
+  log: Array
+  socket
+}
+
+const defaultMQTTRequest: MQTTRequest = {
+  endpoint: "wss://test.mosquitto.org:8081",
+}
+
+const defaultMQTTSession: MQTTSession = {
+  request: defaultMQTTRequest,
+  connectionState: false,
+  connectingState: false,
+  subscriptionState: false,
+  socket: null,
+  log: [],
+}
+
+const dispatchers = defineDispatchers({
+  setRequest(_, { newRequest }: { newRequest: MQTTRequest }) {
+    return {
+      request: newRequest,
+    }
+  },
+  setEndpoint(_, { newEndpoint }: { newEndpoint: string }) {
+    return {
+      request: {
+        endpoint: newEndpoint,
+      },
+    }
+  },
+  setSocket(_, { socket }) {
+    return {
+      socket,
+    }
+  },
+  setConnectionState(_, { state }: { state: boolean }) {
+    return {
+      connectionState: state,
+    }
+  },
+  setConnectingState(_, { state }: { state: boolean }) {
+    return {
+      connectingState: state,
+    }
+  },
+  setSubscriptionState(_, { state }: { state: boolean }) {
+    return {
+      subscriptionState: state,
+    }
+  },
+  setLog(_, { log }) {
```

Comment:
Haven't specified the type for the second parameter.

What is `log` supposed to be here ?